lib/cocoapods/resolver.rb

```diff
@@ -106,8 +106,8 @@ def resolver_specs_by_target
           resolver_specs_by_target[target] = vertices.
             map do |vertex|
               payload = vertex.payload
-              non_library = (!explicit_dependencies.include?(vertex.name) || payload.non_library_specification?) &&
-                (vertex.recursive_predecessors & vertices).all? { |v| !explicit_dependencies.include?(v.name) || v.payload.non_library_specification? }
+              non_library = payload.non_library_specification? || (!explicit_dependencies.include?(vertex.name) &&
```

Comment:
resolver specs for this change

Reply:
the Boolean logic changed here, before the first two conditions were grouped together in the `||`, but now the `&&` is paired with the `!explicit_dependencies.include?` check. was this intentional?

Reply:
this change was intentional -- with the old way, the test spec's dependency on the app spec would cause the _app spec_ to be marked as not `non_library`